src/main/java/com/puppycrawl/tools/checkstyle/site/PropertiesMacro.java

```diff
@@ -0,0 +1,493 @@
+        new File(Paths.get(MAIN_FOLDER_PATH,
+                "api", "AbstractFileSetCheck.java").toString()),
+        new File(Paths.get(MAIN_FOLDER_PATH,
+                CHECKS, "header", "AbstractHeaderCheck.java").toString())
+    );
+
+    /** Mao of all parent modules properties and their javadocs. */
+    private static final Map<String, DetailNode> PARENT_MODULES_PROPERTIES_JAVADOCS =
+            new HashMap<>();
+
+    /** The name of the current module being processed. */
+    private static String currentModuleName = "";
+
+    @Override
+    public void execute(Sink sink, MacroRequest request) throws MacroExecutionException {
+        // until https://github.com/checkstyle/checkstyle/issues/13426
+        if (!(sink instanceof XdocSink)) {
+            throw new MacroExecutionException("Expected Sink to be an XdocSink.");
+        }
+
+        // lazy initialization
+        if (PARENT_MODULES_PROPERTIES_JAVADOCS.isEmpty()) {
+            processParentModules();
+        }
+
+        final String modulePath = (String) request.getParameter("modulePath");
+        final File moduleFile = new File(modulePath);
+        currentModuleName = CommonUtil.getFileNameWithoutExtension(moduleFile.getName());
+
+        processModule(currentModuleName, moduleFile);
+        writePropertiesTable((XdocSink) sink);
+    }
+
+    /**
+     * Collect the properties setters javadocs of the parent modules.
+     *
+     * @throws MacroExecutionException if an error occurs during processing.
+     */
+    private static void processParentModules() throws MacroExecutionException {
+        for (File parentModule : PARENT_MODULES_WITH_PROPERTIES) {
+            final String parentModuleName = CommonUtil
+                    .getFileNameWithoutExtension(parentModule.getName());
+            processModule(parentModuleName, parentModule);
+            final Map<String, DetailNode> parentModuleJavadocs =
+                    ClassAndPropertiesSettersJavadocScraper.getJavadocs();
+            PARENT_MODULES_PROPERTIES_JAVADOCS.putAll(parentModuleJavadocs);
+        }
+    }
+
+    /**
+     * Scrape the Javadocs of the class and its properties setters with
+     * ClassAndPropertiesSettersJavadocScraper.
+     *
+     * @param moduleName the name of the module.
+     * @param moduleFile the module file.
+     * @throws MacroExecutionException if an error occurs during processing.
+     */
+    private static void processModule(String moduleName, File moduleFile)
+            throws MacroExecutionException {
+        if (!moduleFile.isFile()) {
+            final String message = String.format(Locale.ROOT,
+                    "File %s is not a file. Please check the 'modulePath' property.", moduleFile);
+            throw new MacroExecutionException(message);
+        }
+        ClassAndPropertiesSettersJavadocScraper.clear();
+        ClassAndPropertiesSettersJavadocScraper.setModuleName(moduleName);
+        final Checker checker = new Checker();
```

Comment:
please recheck it that is easy to make ClassAndPropertiesSettersJavadocScraper be created for each processModule as new class, it will be a way easier to manage it in future refactoring.

Reply:
I was following the workings of the metadata scraper.

https://github.com/checkstyle/checkstyle/blob/c789b7dedc8823a64481da8bf48c297dbb7a2261/src/main/java/com/puppycrawl/tools/checkstyle/meta/MetadataGeneratorUtil.java#L57-L79

I don't mind changing to an instance-based scraping as long as you can point me to how this whole configuration would look like because I couldn't find what to do.

Reply:
I don't think we do it anywhere, even in main. https://github.com/checkstyle/checkstyle/blob/c789b7dedc8823a64481da8bf48c297dbb7a2261/src/main/java/com/puppycrawl/tools/checkstyle/Main.java#L393

@romani There is no way to provide TreeWalker an instance of a check. It is completely controlled by a configuration, which is a string class name.
https://github.com/checkstyle/checkstyle/blob/c789b7dedc8823a64481da8bf48c297dbb7a2261/src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java#L133